mirgecom/simutil.py

```diff
@@ -434,6 +435,21 @@ def create_parallel_grid(comm, generate_grid):
     return generate_and_distribute_mesh(comm=comm, generate_mesh=generate_grid)
 
 
+def get_reasonable_memory_pool(ctx, queue):
+    """Return an SVM or buffer memory pool based on what the device supports."""
+    from pyopencl.characterize import has_coarse_grain_buffer_svm
+    import pyopencl.tools as cl_tools
+
+    if has_coarse_grain_buffer_svm(queue.device) and hasattr(cl_tools, "SVMPool"):
+        logger.info("Using SVM-based memory pool")
+        return cl_tools.SVMPool(cl_tools.SVMAllocator(  # pylint: disable=no-member
+            ctx, alignment=0, queue=queue))
+    else:
+        from warnings import warn
+        warn("No SVM support, returning a CL buffer-based memory pool")
```

Comment:
Tell the user what aspect is missing (SVM for the device, vs missing newfangled SVM in pyopencl).

Reply:
Done in [9bc4e69](https://github.com/illinois-ceesd/mirgecom/pull/706/commits/9bc4e69bb73ad0a5e938c2f38e6ef68dd235680a)